identity-aware-proxy/IAPClientTest/IAPFixture.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) 2020 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+// 
+// https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System;
+using Xunit;
+
+namespace IAP.Samples.Tests
+{
+    [CollectionDefinition(nameof(IAPFixture))]
+    public class IAPFixture : ICollectionFixture<IAPFixture>
+    {
+        public string CredentialsFilePath { get; }
+        public string IapClientId { get; }
+        public string IapUrl { get; }
+        public string IapTokenExpectedAudience { get; }
+
+        public IAPFixture()
+        {
+            CredentialsFilePath = GetEnvVarOrThrow("GOOGLE_APPLICATION_CREDENTIALS");
+            IapClientId = GetEnvVarOrThrow("TEST_IAP_CLIENT_ID");
+            IapUrl = GetEnvVarOrThrow("TEST_IAP_URI");
+            // The IAP token target audience is of the form /projects/<projectID>/apps/<iap-app-name>
+            IapTokenExpectedAudience = GetEnvVarOrThrow("TEST_IAP_EXPECTED_AUDIENCE");
+        }
+
+        private string GetEnvVarOrThrow(string envVarName) =>
```

Comment:
Presumably having this here is to avoid needing a common library to rely on? I can see that being good for the samples themselves, but perhaps a common *testing* library would be useful?

Reply:
Yes, one of the things I'll do as part of samples revamp is a base fixture class where I can include all these kinds of things.
There's already a testutils library with some functionality in it (retries mostly) that I might just include on that base fixture class.